src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -185,45 +186,72 @@ private void setFilingApiData(
+
+      var publicPrivateDataCombiner = new PublicPrivateDataCombiner(publicDataRetrievalService,
+              privateDataRetrievalService, salt);
+      var publicPrivateOeData = publicPrivateDataCombiner.buildMergedOverseasEntityDataPair(
+              companyNumber, passThroughTokenHeader);
+      var publicPrivateBoData = publicPrivateDataCombiner.buildMergedBeneficialOwnerDataMap(
+              companyNumber, passThroughTokenHeader);
+      var publicPrivateMoData = publicPrivateDataCombiner.buildMergedManagingOfficerDataMap(
+              companyNumber, passThroughTokenHeader);
+
+      ApiLogger.infoContext("PublicPrivateDataCombiner",
+              publicPrivateDataCombiner.logCollatedData());
+
+      updateSubmission.getChanges()
+              .addAll(overseasEntityChangeService.collateOverseasEntityChanges(publicPrivateOeData,
+                      submissionDto));
+      updateSubmission.getChanges()
+              .addAll(beneficialOwnerChangeService.collateBeneficialOwnerChanges(
+                      publicPrivateBoData, submissionDto));
+      updateSubmission.getAdditions()
+              .addAll(beneficialOwnerAdditionService.beneficialOwnerAdditions(submissionDto));
+      updateSubmission.getCessations()
+              .addAll(beneficialOwnerCessationService.beneficialOwnerCessations(submissionDto,
+                      publicPrivateBoData, logMap));
+      updateSubmission.getChanges()
+              .addAll(managingOfficerChangeService.collateManagingOfficerChanges(
+                      publicPrivateMoData, submissionDto));
+      updateSubmission.getCessations()
+              .addAll(managingOfficerCessationService.managingOfficerCessations(submissionDto,
+                      publicPrivateMoData, logMap));
+      updateSubmission.getAdditions()
+              .addAll(managingOfficerAdditionService.managingOfficerAdditions(submissionDto));
+
+      ApiLogger.debug("Updates have been collected", logMap);
+    }
   }
 
   private void setUpdateSubmissionData(
-          Map<String, Object> data, UpdateSubmission updateSubmission, Map<String, Object> logMap) {
+          Map<String, Object> data,
+          UpdateSubmission updateSubmission,
+          boolean isNoChange,
+          Map<String, Object> logMap) {
     data.put(UPDATE_ENTITY_NUMBER_FIELD, updateSubmission.getEntityNumber());
     data.put(UPDATE_TYPE_FIELD, updateSubmission.getType());
     data.put(UPDATE_USER_SUBMISSION_FIELD, updateSubmission.getUserSubmission());
-    data.put(UPDATE_DUE_DILIGENCE_FIELD, updateSubmission.getDueDiligence());
+    if (!isNoChange) {
+      data.put(UPDATE_DUE_DILIGENCE_FIELD, updateSubmission.getDueDiligence());
+    }
     data.put(UPDATE_PRESENTER_FIELD, updateSubmission.getPresenter());
     data.put(FILING_FOR_DATE_FIELD, updateSubmission.getFilingForDate());
     data.put(NO_CHANGES_IN_FILING_PERIOD_FIELD, updateSubmission.getNoChangesInFilingPeriodStatement());
```

Comment:
Does this line need changing at all?

Reply:
I've now removed the noChangesInFilingPeriod from the output JSON. The CHIPS filing consumer can just work off the contents - as CHIPS will need to do anyway.